### PR TITLE
datapath: Remove 2005 route table for IPv4

### PR DESCRIPTION
**Pull request by jschwinger233**
This reverts 3ed62d51d5ac27bbe2c60372850d4ec78c17bb6e for IPv4 part only, as issue #21954 has been resolved by #24208.

Another PR #24882 removes 2005 route table for IPv6.

Signed-off-by: Zhichuan Liang <gray.isovalent.com>

```release-note
Fix broken IPv4 connectivity from outside to NodePort service when using L7 ingress policy, by removing PROXY_RT route table.
```

### CODE REVIEW

**Review by pchaigno — APPROVED** [review]
LGTM.

> This PR removes 2005 route table for IPv6, and changes datapath for return IPv6 traffic from L7 proxy.

Same comment as for the IPv6 PR:
Our release notes usually don't have a subject; it is understood to be "this release". Also nit: the comma before "and" is superfluous since both clauses share the same subject.

File: bpf/init.sh [hunk 1]
```diff
@@ -108,8 +108,7 @@ function move_local_rules()
 
 function setup_proxy_rules()
 {
-	# Any packet from an ingress proxy uses a separate routing table that routes
-	# the packet back to the cilium host device.
+	# TODO(brb): remove $PROXY_RT_TABLE -related code in v1.14
```
Comment (ti-mo) [inline bpf/init.sh]: Is this comment still relevant? This is going into 1.14, so maybe this should be changed to 1.15?

**Comment by julianwiedmann** [comment]
> This reverts https://github.com/cilium/cilium/commit/3ed62d51d5ac27bbe2c60372850d4ec78c17bb6e for IPv4 part only, as issue https://github.com/cilium/cilium/issues/21954 has been resolved by https://github.com/cilium/cilium/pull/24208.

Just for clarity - for IPv4 we don't require https://github.com/cilium/cilium/pull/24208, all that's needed to fix https://github.com/cilium/cilium/issues/21954 is the removal of the 2005 route table (ie. this PR).

**Comment by jschwinger233** [comment]
Just a reminder for whoever wants to run this e2e test locally, now it's required to set env vars `K8S_NODES=3 NO_CILIUM_ON_NODE=k8s3` to run ginkgo so as to test connectivity from outside the cluster.

File: test/k8s/services.go [hunk 2]
```diff
@@ -289,7 +289,7 @@ var _ = SkipDescribeIf(helpers.RunsOn54Kernel, "K8sDatapathServicesTest", func()
 
 			It("Tests NodePort with L7 Policy", func() {
 				applyPolicy(kubectl, demoPolicyL7)
-				testNodePort(kubectl, ni, false, false, 0)
+				testNodePort(kubectl, ni, false, true, 0)
```
Comment (julianwiedmann) [inline test/k8s/services.go]: shouldn't such "from-outside" test be in the `Checks N/S loadbalancing` section below? :thinking:

File: test/k8s/services.go [hunk 3]
```diff
@@ -644,6 +644,28 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`,
...
+				// Same reason as in other L7 test above
+				kubectl.CiliumExecMustSucceedOnAll(context.TODO(),
+					"cilium bpf ct flush global", "Unable to flush CT maps")
+			})
+
+			It("Tests NodePort with L7 Policy from outside", func() {
+				applyPolicy(kubectl, demoPolicyL7)
+				testNodePort(kubectl, ni, false, true, 0)
+			})
+		})
```
Comment (christarazi) [inline test/k8s/services.go:667]: Is this test covered by what's in the CLI connectivity tests or the ci-datapath job?

Comment (jschwinger233) [reply test/k8s/services.go:667]: Just checked cilium-cli, there is connectivity test for reaching nodeport from outside, but no L7 policy is applied there to cover #21954: 
https://github.com/cilium/cilium-cli/blob/91329434f04b605e19eb329b8e891f0aa1df49a0/connectivity/tests/service.go#L208-L217
I understand we plan to abandon e2e test suites, I'll open another PR for cilium-cli to cover this scenario after this gets merged.

**Review by christarazi — APPROVED** [review]

**Review by julianwiedmann — APPROVED** [review]

**Comment by michi-covalent** [comment]
why does this still have `needs-backport/1.13` label? shouldn't it be `backport-pending/1.13` instead since https://github.com/cilium/cilium/pull/25086 is open? 👀

**Comment by margamanterola** [comment]
Regarding the release note, what is the "2005 route table"? Given that this is release-note/bug, we want to give our users a better indication of when they would be affected by this bug and when they wouldn't. With the current description it's very hard.

**Comment by jschwinger233** [comment]
> Regarding the release note, what is the "2005 route table"? 

It's the PROXY_RT_TABLE whose id is 2005, we created this route table to let all return traffic from proxy go to cilium_host, and this table caused #21954.

Thanks for reminding me of the bad release note, I'll revise it for better understanding.

**Pull request merged**